test_driver/app_test.dart

```diff
@@ -0,0 +1,132 @@
+      expect(await driver.getText(text), 'Email');
+    });
+
+    test('verify the password hint text on login screen', () async {
+      SerializableFinder text = find.text('Password');
+      await driver.waitFor(text);
+      expect(await driver.getText(text), 'Password');
+    });
+
+    test('verify the text on login button', () async {
+      final buttonFinder = find.text('Log in');
+      expect(await driver.getText(buttonFinder), 'Log in');
+    });
+
+    test(
+      'verify the text input in textfields',
+      () async {
+        String emailValue = 'abcd@gmail.com';
+        SerializableFinder emailTextField =
+            find.byValueKey('emailTextFieldKey');
+        await driver.waitFor(emailTextField);
+        await driver.tap(emailTextField);
+        await driver.enterText(emailValue);
+        SerializableFinder emailText = find.text(emailValue);
+        await driver.waitFor(emailText);
+        expect(await driver.getText(emailText), emailValue);
+
+        String passwordValue = '12345678910';
+        SerializableFinder passwordTextField =
+            find.byValueKey('passwordTextFieldKey');
+        await driver.waitFor(passwordTextField);
+        await driver.tap(passwordTextField);
+        await driver.enterText(passwordValue);
+        SerializableFinder passwordText = find.text(passwordValue);
+        await driver.waitFor(passwordText);
+        expect(await driver.getText(passwordText), passwordValue);
+      },
+      timeout: const Timeout(
+        Duration(minutes: 2),
+      ),
+    );
+
+    test(
+      'verify login error message',
+      () async {
+        String emailValue = 'abcd@gmail.com';
+        SerializableFinder emailTextField =
+            find.byValueKey('emailTextFieldKey');
+        await driver.waitFor(emailTextField);
+        await driver.tap(emailTextField);
+        await driver.enterText(emailValue);
+        SerializableFinder emailText = find.text(emailValue);
+        await driver.waitFor(emailText);
+
+        String passwordValue = '12345678910';
+        SerializableFinder passwordTextField =
+            find.byValueKey('passwordTextFieldKey');
+        await driver.waitFor(passwordTextField);
+        await driver.tap(passwordTextField);
+        await driver.enterText(passwordValue);
+        SerializableFinder passwordText = find.text(passwordValue);
+        await driver.waitFor(passwordText);
+
+        SerializableFinder loginButton = find.byValueKey('loginButtonKey');
+        await driver.waitFor(loginButton);
+        await driver.tap(loginButton);
+
+        String errorMessage = 'Bad request';
+        SerializableFinder loginErrorText = find.text(errorMessage);
+        await driver.waitFor(loginErrorText);
+        expect(await driver.getText(loginErrorText), errorMessage);
+      },
+      timeout: const Timeout(
+        Duration(minutes: 2),
+      ),
+    );
+
+    test(
+      'verify login success and navigate to home screen',
```

Comment:
Do you think we should have a test case, which submits the survey form too? 

I think yes, but then we will have to wait until we finish #43

Reply:
Yes, we need to wait for finishing all other PRs.